pkg/broker/work_engine.go

```diff
@@ -50,10 +50,33 @@ func (engine *WorkEngine) StartNewAsyncJob(
 	if token == "" {
 		token = engine.Token()
 	}
-	go work.Run(token, engine.topic(topic))
+	go engine.start(token, work, topic)
```

Comment:
This feels really weird to me. Starting the engine  for each job. Feels like the engine should always be running and waiting for new jobs to be added to it to be run.

Reply:
This could be called startSubscriberListeningAndRun(). I would prefer to keep in smaller maybe something like `startJob`?